### PR TITLE
chore: make usePlatformMe less spammy

### PR DESCRIPTION
**Pull request by ThyMinimalDev**
## What does this PR do?

UsePlatformMe hooks is spamming api v2 unecessarily

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: apps/web/components/settings/platform/hooks/usePlatformMe.ts [hunk 1]
```diff
@@ -6,14 +6,26 @@ export const usePlatformMe = () => {
...
       const response = await fetch(`/api/v2/me`, {
         method: "get",
         headers: { "Content-type": "application/json" },
       });
-      const data = await response.json();
 
-      return data.data as UserResponse;
+      if (response.status === 200) {
+        const data = await response.json();
+        return data.data as UserResponse;
```
Comment (hariombalhara) [inline apps/web/components/settings/platform/hooks/usePlatformMe.ts:25]: Might be a good idea to use zod instead.

File: apps/web/components/settings/platform/hooks/usePlatformMe.ts [hunk 2]
```diff
@@ -6,14 +6,26 @@ export const usePlatformMe = () => {
   const QUERY_KEY = "get-platform-me";
   const platformMeQuery = useQuery<UserResponse>({
     queryKey: [QUERY_KEY],
-    queryFn: async (): Promise<UserResponse> => {
+    retryOnMount: false,
+    staleTime: 300000,
+    retry: (failureCount, error) => {
+      if (error.message === "Internal Server Error") {
```
Comment (hariombalhara) [inline apps/web/components/settings/platform/hooks/usePlatformMe.ts:12]: Better to use a code for this or rely on statusCode being 5xx. Text might start getting translated, causing weird issues.

**Review by hariombalhara — DISMISSED** [review]
LGTM !!

**Review by Ryukemeister — DISMISSED** [review]

File: apps/web/components/settings/platform/hooks/usePlatformMe.ts [hunk 3]
```diff
@@ -6,14 +6,26 @@ export const usePlatformMe = () => {
   const QUERY_KEY = "get-platform-me";
   const platformMeQuery = useQuery<UserResponse>({
     queryKey: [QUERY_KEY],
-    queryFn: async (): Promise<UserResponse> => {
+    retryOnMount: false,
+    staleTime: 300000,
```
Comment (supalarry) [inline apps/web/components/settings/platform/hooks/usePlatformMe.ts:10]: 300000 seconds is 83 hours, so i am wondering if someone goes to dashboard and then buys platform plan will they be able to use platform?

I would reduce this to an hour or something or maybe even 30 minutes.

Comment (supalarry) [reply apps/web/components/settings/platform/hooks/usePlatformMe.ts:10]: and in such a case remove retryOnMount

**Review by keithwillcode — CHANGES_REQUESTED** [review]
Looks like these changes broke a bunch of E2E tests

File: apps/web/components/settings/platform/hooks/usePlatformMe.ts (hunk 1 above)
Comment (zomars) [reply apps/web/components/settings/platform/hooks/usePlatformMe.ts:25]: zod is very heavy for client bundle size. I wouldn't introduce it only for this

**Comment by ThyMinimalDev** [comment]
not sure what's wrong with e2e tests here, putting back in draft until then

**Pull request closed without merging**